src/System.Security.Cryptography.Xml/tests/XmlLicenseEncryptedRef.cs

```diff
@@ -0,0 +1,229 @@
+                    }
+
+                    bool keyMismatch = true;
+                    foreach (AsymmetricAlgorithm key in asymmetricKeys)
+                    {
+                        RSA rsaKey = key as RSA;
+                        Assert.NotNull(rsaKey);
+
+                        rsaInputParams = rsaKey.ExportParameters(false);
+
+                        if (!PublicKeysEqual(rsaParams, rsaInputParams))
+                        {
+                            continue;
+                        }
+
+                        keyMismatch = false;
+
+                        // Decrypt session key
+                        byte[] encryptedKeyValue = encryptedKey.CipherData.CipherValue;
+
+                        if (encryptedKeyValue == null)
+                            throw new CryptographicException("MissingKeyCipher");
+
+                        byte[] decryptedKey = EncryptedXml.DecryptKey(encryptedKeyValue,
+                                                                     rsaKey, true);
+
+                        if (decryptedKey == null)
+                        {
+                            throw new CryptographicException("KeyDecryptionFailure");
+                        }
+
+                        aes = Aes.Create();
+                        aes.Key = decryptedKey;
+                        aes.Padding = PaddingMode.PKCS7;
+                        aes.Mode = CipherMode.CBC;
+                        break;
+                    }
+
+                    if (keyMismatch)
+                    {
+                        throw new Exception("Invalid License - AsymmetricKeyMismatch");
+                    }
+                }
+                else if (clause is KeyInfoName)
+                {
+                    Assert.True(false, "This test should not have KeyInfoName clauses");
+                }
+                else
+                    throw new CryptographicException("MalformedKeyInfoClause");
+
+                break;
+            }
+
+            return DecryptStream(toDecrypt, aes);
+        }
+
+        private static Stream DecryptStream(Stream toDecrypt, SymmetricAlgorithm aes)
+        {
+            Assert.NotNull(aes);
+
+            byte[] IV = new byte[aes.BlockSize / 8];
+
+            // Get the IV from the encrypted content.
+            toDecrypt.Read(IV, 0, IV.Length);
+            byte[] encryptedContentValue = new byte[toDecrypt.Length - IV.Length];
+
+            // Get the encrypted content following the IV.
+            toDecrypt.Read(encryptedContentValue, 0, encryptedContentValue.Length);
+
+            byte[] decryptedContent;
+            ICryptoTransform decryptor = aes.CreateDecryptor(aes.Key, IV);
+
+            using (var msDecrypt = new MemoryStream())
+            using (var csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Write))
+            {
+                csDecrypt.Write(encryptedContentValue, 0, encryptedContentValue.Length);
+                csDecrypt.FlushFinalBlock();
```

Comment:
After you stop disposing msDecrypt and just return it, you can remove this call (because it's automatically called during Dispose)